airflow/www/views.py

```diff
@@ -325,10 +319,36 @@ def get_int_arg(value, default=0):
             paging=wwwutils.generate_pages(current_page, num_of_pages,
                                            search=escape(arg_search_query) if arg_search_query else None,
                                            showPaused=not hide_paused),
-            auto_complete_data=auto_complete_data,
             num_runs=num_runs,
             tags=tags)
 
+    @expose('/dag_autocomplete')
+    @has_access
```

Comment:
Have you tested this with a non-admin role?

Reply:
Aha good spot, it seems I need to either define a new permission for this or piggy back the 'can_index' permission. What would you recommend?

Reply:
Just spoke with @mik-laj on Slack:

"I think it would be better to create a new permission.  I will create the API in the near future and the model of permissions will change radically.
each view is a new permission and this rule will make it easier for us to refactor in the future now."

Reply:
Implemented the above

Reply:
For the record: I am not a fan of each view being a new permission; instead I am of the view that permissions should be about operations/concepts - so can list, can get details etc -> I can see no reason why you might want to view details/list a dag but not have it show up in the auto complete list for instance

(Not saying change this now, but if we re-think the permissions for the API work)

Reply:
And more for future work than anything else: should methods like this live on the DagModelView instead?

Reply:
Yea i guess `dag_stats` too, could potentially break people consuming the end point directly on its existing path mind.

Reply:
Since this isn't an official API, not even in the "/api" namespace I wouldn't object too much to moving it around, but probably not worth moving existing things until we do the larger API work.